generator/internal/client_generator.cc

```diff
@@ -121,7 +123,7 @@ Status ClientGenerator::GenerateHeader() {
   std::unique_ptr<::google::cloud::AsyncStreamingReadWriteRpc<
       $request_type$,
       $response_type$>>
-  Async$method_name$(Options opts = {});
+  Async$method_name$(ExperimentalTag, Options opts = {});
```

Comment:
So we are signing the user up for at least one break. Is that a good thing? We could avoid it by doing like:

```
AsyncFoo(Options opts = {}, ExperimentalTag tag = {});
```

(but then they are less likely to notice the tag)

Reply:
I want this tag to be a *required* parameter, so folks cannot accidentally use these APIs without knowing that they are experimental (I guess they could say `using GA = ExperimentalTag;` but that would be just self-defeating).

To avoid the break we can keep an overload:

```cc
[[deprecated("use non-experimental version")]] Blah AsyncFoo(ExperimentalTag, Options opts = {}) {
 return AsyncFoo(std::move(opts));
 }

Blah AsyncFoo(Options opts);
```

Reply:
Thanks for clarifying.